src/hotspot/cpu/aarch64/aarch64_neon_ad.m4

```diff
@@ -2481,3 +2481,36 @@ instruct vmask_lasttrue16B(iRegINoSp dst, vecX src) %{
   %}
   ins_pipe(pipe_slow);
 %}
+
+instruct vmask_tolong8B(iRegLNoSp dst, vecD src) %{
+  match(Set dst (VectorMaskToLong src));
+  ins_cost(5 * INSN_COST);
+  format %{ "vmask_tolong $dst, $src\t# convert mask to long (8B)" %}
+  ins_encode %{
+    // Input "src" is a vector of boolean represented as
+    // bytes with 0x00/0x01 as element values.
+
+    __ fmovd(as_Register($dst$$reg), as_FloatRegister($src$$reg));
+    __ bytemask_compress(as_Register($dst$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vmask_tolong16B(iRegLNoSp dst, vecX src, iRegL tmp) %{
```

Comment:
"iRegL tmp" should be "iRegLNoSp tmp" or use rscratch directly?

Reply:
Yes, good catch. This kind of bug is horrible to detect.

Reply:
Thanks. It's fixed now.